### ISSUE TITLE
Unknown metaclass superclass causes conflicting-metaclass, propagated further in 0.0.80

### ISSUE THREAD

**Issue by jelle-openai** [astral-sh/ty#4510]
### Summary

The underlying false positive also exists in 0.0.79; 0.0.80 additionally reports it on a downstream subclass.

```python
from typing import Any

capybara: Any = type

class Agouti(capybara):
    pass

class Hutia(object, metaclass=Agouti):  # ty: ignore[conflicting-metaclass]
    pass

class Gundi(Hutia, object):
    pass
```

The ignore suppresses an existing diagnostic on `Hutia` in **both** versions. With that ignore, ty 0.0.79 passes, while 0.0.80 reports:

```text
error[conflicting-metaclass]: The metaclass of a derived class (`Gundi`) must be a subclass of the metaclasses of all its bases, but `Agouti` (metaclass of base class `Hutia`) and `type` (metaclass of base class `object`) have no subclass relationship
```

Without the ignore, 0.0.79 reports one conflict (`Hutia`) and 0.0.80 reports two (`Hutia` and `Gundi`). The ignore is only there to isolate the version difference.

The code runs successfully on Python 3.12. At runtime `Agouti` derives from `type`, and both `Hutia` and `Gundi` have metaclass `Agouti`. Replacing `capybara: Any = type` with `capybara = type` makes both versions accept both classes without any ignores.

Expected: an unknown superclass should not be sufficient evidence that the metaclass is incompatible. At least, it would be useful to avoid propagating this conflict to users of third-party classes when the original problem is inside a dependency.

The added downstream error appears consistent with the deliberate metaclass-candidate preservation in https://github.com/astral-sh/ruff/pull/28461. I have not bisected it. 

Real world context: this arose with a class inheriting from a Pydantic metaclass that is not accessible to the type checker.

Versions tested:
- ty 0.0.79 (b4cd79275 2026-09-07)
- ty 0.0.80 (7fd8e1569 2026-09-09)

**Comment by charliermarsh** [comment]
https://github.com/astral-sh/ruff/pull/28461 leads to the _propagated_ error (though isn't the root cause of the issue).
